src/Denner/Client/Response/BaseResponse.php

```diff
@@ -50,7 +50,17 @@ public function toArray()
     protected function getData()
     {
         try {
-            $data = $this->getHttpResponse()->json() ?: array();
+//            switch ($this->getHttpResponse()->getHeader('Content-Type')) {
+//                case 'application/xml; charset=utf-8':
+//                case 'application/xml':
+//                    $data = $this->xml2array($this->getHttpResponse()->xml());
+//                    break;
+//                case 'application/json; charset=utf-8':
+//                case 'application/json':
+//                default:
+                    $data = $this->getHttpResponse()->json() ?: array();
+//                    break;
+//            }
```

Comment:
Why did you add this and then comment it?

Reply:
Initially, I had to set the Accept header to 'application/xml' for requests, so that the response Content-Type was XML to parse the token authorization answer (that is play text on JSON, but correct with XML).

But the short xml2array function I implemented here is not good at all, should use a professional library/deserialzer for this.

Only in a second pass I decided to introduce a new Response\PlainTextResponse to bypass the incorrect response returned by Magento 2, so that I could set the Accept header back to JSON.

But the extension to accept XML wold be a correct enhancement for this library.